lib/core/utils/get-flattened-tree.js

```diff
@@ -136,16 +151,23 @@ function flattenTree(node, shadowId, parent) {
  * excluding comment nodes and the shadow DOM nodes <content> and <slot>
  *
  * @param {Node} [node=document.documentElement] optional node. NOTE: passing in anything other than body or the documentElement may result in incomplete results.
- * @param {String} [shadowId] optional ID of the shadow DOM that is the closest shadow
- *                           ancestor of the node
+ * @param {String} [shadowId] optional ID of the shadow DOM that is the closest shadow ancestor of the node
  */
 function getFlattenedTree(node = document.documentElement, shadowId) {
   cache.set('nodeMap', new WeakMap());
 
+  // reset selectorMap for each call of getFlattnedTree so each tree
+  // can have it's own cache
+  cache.set('selectorMap', {});
+  cache.get('nodeIndex', 0);
+
   // specifically pass `null` to the parent to designate the top
   // node of the tree. if parent === undefined then we know
   // we are in a disconnected tree
-  return flattenTree(node, shadowId, null);
+  const tree = flattenTree(node, shadowId, null);
+  tree[0]._selectorMap = cache.get('selectorMap');
```

Comment:
Where is this cache value coming from? It isn't set in this function. How does anyone know what this, and how can anyone trust that this exists at all?